src/pineko/check.py

```diff
@@ -167,34 +186,50 @@ def contains_ren(grid, max_as, max_al):
         is ren scale-variation available for as
     bool
         is ren scale-variation available for al
+    bool
+        is the central as order reached by the grid
+    bool
+        is the central al order reached by the grid
     """
     order_array = np.array([order.as_tuple() for order in grid.orders()])
     order_mask = pineappl.grid.Order.create_mask(grid.orders(), max_as, max_al)
     order_list = order_array[order_mask]
     as_orders = []
     al_orders = []
+    min_al = min([ord[1] for ord in order_list])
     for order in order_list:
-        as_orders.append(order[0])
-        al_orders.append(order[1])
-    # ren sv starts one order after the first order with as
-    min_as = 1 if min(as_orders) == 0 else min(as_orders)
-    # ren sv starts one order after the first order with al
-    min_al = 1 if min(al_orders) == 0 else min(al_orders)
+        if order[1] == min_al:
+            as_orders.append(order)
+        if order[1] != 0:
+            al_orders.append(order)
+    min_as = min([ord[0] for ord in as_orders], default=0)
     order_as_is_present = False
     order_al_is_present = False
     sv_as_present = False
     sv_al_present = False
-    for order in order_list:
-        if order[0] == min_as + 1:
+    # ren sv starts one order after the first order with as (al) if the first order in aa (al) is not as^0 (al^0)
+    add_order_as = 1 + (max_as - 2)
+    add_order_al = 1 + (max_al - 2)
+    if max_as == 1:
+        add_order_as = 1
+    if max_al == 1:
+        add_order_al = 1
+    for order in as_orders:
+        if order[0] == min_as + add_order_as:
             order_as_is_present = True
             if order[-2] != 0:
                 sv_as_present = True
-        if order[1] == min_al + 1:
+    for order in al_orders:
+        if order[1] == min_al + add_order_al:
             order_al_is_present = True
             if order[-2] != 0:
                 sv_al_present = True
+    if min_as == 0 and max_as == 2:
+        sv_as_present = True
+    if min_al == 0 and max_al == 2:
+        sv_al_present = True
     if not order_as_is_present:
         sv_as_present = True
     if not order_al_is_present:
         sv_al_present = True
-    return sv_as_present, sv_al_present
+    return sv_as_present, sv_al_present, order_as_is_present, order_al_is_present
```

Comment:
Same as `contains_fact`: it is overly complicated for no reason.

Split it, and possibly reuse pieces. Most likely, most of `contains_fact` and `contains_ren` is just the same.